### DISCUSSION TITLE
Remote ID text field is limited to 21 characters on the linux client!

### DISCUSSION THREAD

**Discussion by nullbyto** [rustdesk/rustdesk#5768] [General]
### Bug Description

I have setup `Tailscale` VPN for my devices, and I connect to them using the internal `DNS` system in `Tailscale` instead of typing the IP via direct IP. The syntax used is `<hostname>:<port>`, but unfortunately the characters allowed to be typed in the text field of the application is limited to 21 characters only on the `Linux` client, which isn't enough for the domain name and port needed to connect.
On the `windows` and `android` clients it works normally.

### How to Reproduce

Run RustDesk on a linux machine, and just start typing in the `Remote ID` text field, it will stop to the length 21 characters only.

### Expected Behavior

The text field for the Remote ID shouldn't be limited to such a short string length like 21.

### Operating system(s) on local side and remote side

Linux (NixOS) -> Windows 10

### RustDesk Version(s) on local side and remote side

1.2.2 -> 1.2.2

### Screenshots

![image](https://github.com/rustdesk/rustdesk/assets/67614151/fc161d7f-aa5f-44f5-bbe8-60b827bc2ea4)

### Additional Context

EDIT: I am using NixOS linux.

**Comment by rustdesk** [comment]
I can not see this limit on my Linux. BTW, windows/linux share the same code, no way to limit linux seperately.

**Reply by nullbyto** [reply]
It works on your client, that is really interesting! I thought it has different code since the frontend looks different on both devices.

Anyways, after i have searched in the code a little i have found this line in the code of the ui indicating a maxlength of 21 for that text field in the file `src/ui/index.tis`.

```
return <input type="text" #remote_id .outline-focus novalue={translate("Enter Remote ID")} maxlength="21"
```
 https://github.com/rustdesk/rustdesk/blob/a1c910e3aa65dc78c0ccd9273feb651865e92f64/src/ui/index.tis#L1012

**Reply by nullbyto** [reply]
I might add, that this issue happens only on my NixOS machine but not on Archlinux! Eventho both have 1.2.2 version, but the frontend looks a little different for some reason.
The settings looks like this on the NixOS machine:
![image](https://github.com/rustdesk/rustdesk/assets/67614151/569187ca-970a-4d61-bcb7-754b40b2e328)

**Reply by nullbyto** [reply]
For some reason it is only related to NixOS, I assume, the package in nixpkgs is packaged with another frontend code. 
The only easiest workaround I found for now is to install `rustdesk` using flatpak on NixOS, for anyone having the same problem. 

I still don't have an explanation for the code I pointed out above, it is not a coincidence with the number 21.

**Comment by SergeyMy** [comment]
This is a Sciter branch, it is slowly dying out, use Flutter

**Reply by nullbyto** [reply]
Thanks for your feedback, ya it seems in the NixOS version its using that version.
